meta/saiserialize.c

```diff
@@ -1224,15 +1336,42 @@ int sai_serialize_attribute(
 
     buf += sprintf(buf, "\"value\":");
 
-    ret = sai_serialize_attribute_value(buf, meta, &attribute->value);
-
-    if (ret < 0)
+    if (NULL != meta)
     {
-        SAI_META_LOG_WARN("failed to serialize attribute value");
-        return SAI_SERIALIZE_ERROR;
+        ret = sai_serialize_attribute_value(buf, meta, &attribute->value);
+        if (ret < 0)
+        {
+            serialize_binary = true;
+        }
+        else
+        {
+            buf += ret;
+        }
+    }
+    else
+    {
+        serialize_binary = true;
+    }
+    if (serialize_binary)
+    {
+        buf += sprintf(buf, "{");
+        uint32_t idx;
+        for (idx = 0; idx < sizeof(attribute->value); idx++)
+        {
+            if (idx != 0)
+            {
+                buf += sprintf(buf, ",");
+            }
+            ret = sai_serialize_uint8(buf, ((const uint8_t*)&attribute->value)[idx]);
+            if (ret < 0)
+            {
+                SAI_META_LOG_WARN("failed to serialize attribute value index %u", idx);
+                return SAI_SERIALIZE_ERROR;
+            }
+            buf += ret;
+        }
+        buf += sprintf(buf, "}");
```

Comment:
please improve your code quality, take a look at the file how lines are written, add empty line before each IF and after each }